codejail.profile

```diff
@@ -0,0 +1,100 @@
+# AppArmor profile for running codejail-service in devstack.
+#
+#                         #=========#
+#                         # WARNING #
+#                         #=========#
+#
+# This is not a complete and secure apparmor profile! Do not use this
+# in any deployed environment (even a staging environment) without
+# careful inspection and modification to fit your needs.
+#
+# Failure to apply a secure apparmor profile *will* likely result in a
+# compromise of your environment by an attacker.
+#
+# We may at some point make this file good enough for confinement in
+# production, but for now it is only intended to be used in devstack.
+
+# Sets standard variables used by abstractions/base, later. Controlled
+# by OS, see /etc/apparmor.d/tunables/global for contents.
+include <tunables/global>
+
+# Require that the system understands the feature set that this policy was written
+# for. If we didn't include this, then on Ubuntu >= 22.04, AppArmor might assume
+# the wrong feature set was requested, and some rules might become too permissive.
+# See https://github.com/netblue30/firejail/issues/3659#issuecomment-711074899
+abi <abi/3.0>,
+
+# This outer profile applies to the entire container, and isn't as
+# important. If the sandbox profile doesn't work, it's not likely that
+# the outer one is going to help. But there may be some small value in
+# defense-in-depth, as it's possible that a bug in the child (sandbox)
+# profile isn't present in the outer one.
+profile codejail_service flags=(mediate_deleted) {
```

Comment:
1. [proposal] I think using the name `codejail_service_profile` will make this more clear throughout other code and docs for those that are less familiar.
2. I wanted to know what `mediate_deleted` is for before I got to the answer. Maybe add something like: "See the child profile below for an explanation of the `mediate_deleted` flag."

Reply:
I'd really like to avoid having the word "profile" in the profile name. It would be like having "variable" in a variable name.

Yeah, I can make a forward reference to the later comment.

Reply:
I'm going to make one more push for the name change. 

First, there is a reason that most of your english comments refer to "child profile", because the word "child" isn't meaningful enough on its own.

Second, I'm more concerned with the uses of the profile name than the declaration. For example:
```
signal (send) set=(kill) peer=codejail_service//child,
```

For those that don't know this syntax well, this is not as clear as:
```
signal (send) set=(kill) peer=codejail_service//child_profile,
```

Additionally, if you decide to rename "child" to "sandbox", the following gets even more ambiguous for those that aren't expert in this syntax:
```
signal (send) set=(kill) peer=codejail_service//sandbox,
```

Reply:
The comment there currently says "Allow sending a kill signal to the sandbox ..." -- what if I change that to "Allow sending a kill signal to the `child` subprofile ..."? This would indicate what the `//` syntax means, and use its name explicitly.

(I still might also want to rename that inner profile. Maybe `jailed`?)

Reply:
I guess you really don't like the idea of using `_profile`. 😉 

Regarding possible renames, what about `codejail_sandbox`?

Reply:
Yup, very much don't want it, if at all possible.

`codejail_sandbox` sounds fine!